mtags/src/main/scala-2/scala/meta/internal/pc/InferredTypeProvider.scala

```diff
@@ -67,16 +72,30 @@ final class InferredTypeProvider(
       start + identLength + backtickInc
     }
 
+    def removeType(nameEnd: Int, tptEnd: Int) = {
+      val txt = params.text()
+      txt.substring(0, nameEnd) +
+        txt.substring(tptEnd + 1, params.text().length())
+    }
+
     typedTree match {
       /* `val a = 1` or `var b = 2`
        *     turns into
        * `val a: Int = 1` or `var b: Int = 2`
        */
-      case vl @ ValDef(_, name, tpt, _) if !vl.symbol.isParameter =>
+      case vl @ ValDef(_, name, tpt, rhs) if !vl.symbol.isParameter =>
         val nameEnd = findNameEnd(tpt.pos.start, name)
         val nameEndPos = tpt.pos.withEnd(nameEnd).withStart(nameEnd).toLSP
+        adjustedEndPos.foreach(nameEndPos.setEnd(_))
         val typeNameEdit = new TextEdit(nameEndPos, ": " + prettyType(tpt.tpe))
```

Comment:
```suggestion
        lazy val typeNameEdit = new TextEdit(nameEndPos, ": " + prettyType(tpt.tpe))
```

or calculate this in `else` branch

Reply:
Fixed!